packages/jellyfish-api-core/__tests__/category/loan/listCollateralTokens.test.ts

```diff
@@ -56,11 +50,89 @@ describe('Loan listCollateralTokens', () => {
     }])
     await testing.generate(1)
 
-    const data = await testing.rpc.loan.listCollateralTokens()
-    expect(data).toStrictEqual({
-      // activateAfterBlock = 122, which is the next block after setcollateraltoken executed
-      [collateralTokenId1]: { activateAfterBlock: new BigNumber(122), factor: new BigNumber(0.5), priceFeedId: 'AAPL/USD', token: 'AAPL' },
-      [collateralTokenId2]: { activateAfterBlock: new BigNumber(130), factor: new BigNumber(1), priceFeedId: 'TSLA/USD', token: 'TSLA' }
-    })
+    {
+      // List collateral tokens that are activated
+      const data1 = await testing.rpc.loan.listCollateralTokens()
+      const data2 = await testing.rpc.loan.listCollateralTokens({ all: false })
+      expect(data1).toStrictEqual(data2)
+      expect(data1).toStrictEqual({
+        [collateralTokenId1]: {
+          activateAfterBlock: new BigNumber(blockCount),
+          factor: new BigNumber(0.5),
+          priceFeedId: 'AAPL/USD',
+          token: 'AAPL'
+        }
+      })
+    }
+
+    {
+      // List all collateral tokens
+      const data = await testing.rpc.loan.listCollateralTokens({ all: true })
+      expect(data).toStrictEqual({
+        [collateralTokenId1]: {
+          activateAfterBlock: new BigNumber(blockCount),
+          factor: new BigNumber(0.5),
+          priceFeedId: 'AAPL/USD',
+          token: 'AAPL'
+        },
+        [collateralTokenId2]: {
+          activateAfterBlock: new BigNumber(130),
+          factor: new BigNumber(1),
+          priceFeedId: 'TSLA/USD',
+          token: 'TSLA'
+        }
+      })
+    }
+  })
+})
+
+describe('Loan listCollateralTokens with height', () => {
```

Comment:
This tests, add another collateral token with height. then show what it returns with `height`.
Also can add a test to show that if `all` and `height` combination used, only `all` is considered.

Reply:
Fixed